dask_sql/physical/rel/custom/create_table_as.py

```diff
@@ -39,6 +39,10 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
         create_memory_table = rel.create_memory_table()
 
         schema_name, table_name = context.schema_name, create_memory_table.getName()
+        if "." in table_name:
+            raise KeyError(
+                f"Cannot create table from {table_name} due to schema error."
+            )
```

Comment:
This logic basically makes it impossible to create a table when the schema is specified.

Without it, if we were using schema `foo` and tried to create table `other.bar` (regardless of whether schema `other` exists), `other.bar` is added to `foo`, but then it is impossible to access it (e.g., with `select`) because we cannot parse `foo.other.bar`.

Since this test doesn't cover `create table [schema].[name] ...`, I just made it error regardless of whether or not a valid schema is specified.